### PR TITLE
LibJS: Add spec comments to RegExp.prototype and fix a couple bugs

### PR DESCRIPTION
**Pull request by trflynn89**
Not a particularly glorious PR, but I was wondering if any of the RegExp.prototype methods had normative changes, which I think is just easier with spec comments. This did find a couple of bugs though:
* A `vm.throw_exception` invocation that should have been a `return vm.throw_completion`.
* Some non-spec `to_object` invocations (one of which was masking the above bug).
* Using `TRY` in places that should be using `MUST`.

### CODE REVIEW

File: Userland/Libraries/LibJS/Runtime/RegExpPrototype.cpp [hunk 1]
```diff
@@ -278,147 +431,186 @@ ThrowCompletionOr<Value> regexp_exec(GlobalObject& global_object, Object& regexp
...
+        MUST(array->create_data_property_or_throw(n, js_string(vm, match_str)));
 
-        if (match_str.is_empty())
-            TRY(increment_last_index(global_object, *regexp_object, string.view(), unicode));
+        // 3. If matchStr is the empty String, then
+        if (match_str.is_empty()) {
+            // a. Let thisIndex be ℝ(? ToLength(? Get(rx, "lastIndex"))).
+            // b. Let nextIndex be AdvanceStringIndex(S, thisIndex, fullUnicode).
+            // c. Perform ? Set(rx, "lastIndex", 𝔽(nextIndex), true).
+            TRY(increment_last_index(global_object, *regexp_object, string.view(), full_unicode));
```
Comment (davidot) [inline Userland/Libraries/LibJS/Runtime/RegExpPrototype.cpp:564]: Perhaps these steps should be moved into the `increment_last_index` function with a comment here that steps 3.a ... 3.c are executed in `increment_last_index`?
Or maybe just adding the steps to `increment_last_index` not sure what would be nicer

**Review by davidot — APPROVED** [review]
I couldn't find any issues just one small nitpick

File: Userland/Libraries/LibJS/Runtime/RegExpPrototype.cpp (hunk 1 above)
Comment (trflynn89) [reply Userland/Libraries/LibJS/Runtime/RegExpPrototype.cpp]: Sure, I move the comments to `increment_last_index` and referred to them in the call sites

**Review by linusg — APPROVED** [review]
Nice work, LGTM!

**Pull request merged**